README.md

```diff
@@ -1,70 +1,120 @@
+"dob": "Date",          // user's date of birth
+"address": "String",    // user's address
+"created_at": "Date",   // record created time
+"modified_at": "Date"   // record modified time
+}
+```
 
+The `user_name`, `password`, `salt` and `modified_at`are added beyond the original test requirement, to complete the login/logout task and complete the infos of tracking in future.
 
-## What We Care About
+#### API
 
-Feel free to use any libraries you would use if this were a real production app, but remember we're interested in your code & the way you solve the problem, not how well you can use a particular library.
+| Method | URI                     | Description                              |
+| :----: | :---------------------- | :--------------------------------------- |
+|  GET   | /api/users/{id}         | Get user by id                           |
+|  POST  | /api/users/             | Create new user                          |
+|  PUT   | /api/users/{id}         | Update User by id (user can only edit their slef data after login) |
+| DELETE | /api/users/{id}         | Delete user by id                        |
+|  POST  | /api/users/{id}/replace | Same to PUT /api/users/{id} (Loopback framework effect) |
+|  POST  | /api/users/login        | Login specific user with `user_name` and `password` |
+|  POST  | /api/users/logout       | Logout currently logined user(base on accessToken) |
 
-We're interested in your method and how you approach the problem just as much as we're interested in the end result.
+## Notes
 
-Here's what you should aim for:
+Loopback has build in models can easily complete the task in this test.
 
-- Good use of current restful api design & performance best practices.
+* Loopback have build in `User` model, with username, email, password and etc.
+* Loopback have build in `AccessToken` model, to record the users logined info.
 
-- Solid testing approach
+Loopback allow to inherit from other models and overwrite the existing method to do polymorphism.
 
-- Extensible code;
+To me, i think the implementation with the build-in User and AccessToken are kind of cheat for this test, so all the model in this test are based on loopback's `PersistedModel` which is the base model for loopback other models.
 
-## Q&A
+In this test during login part i use _Secure Salted Password Hashing_ to encrypted user's password in database, and use [uuid](https://github.com/kelektiv/node-uuid) to generate the logined user's token to verify, the verification has borrow some of the loopback lib code to make it can works in Loopback framework way.
 
-* Where should I send back the result when I'm done?
+Beside this i did the following to make the code be easier to expand and more friendly to end users.
 
-Fork this repo and send us a pull request when you think you are done. We don't have deadline for the task.
+* Create based model `WBase` to deal with general meta infos like `created_at` and `modified_at`
```

Comment:
Yes, the inheritance is the original idea that i used form the loopback framework, for Loopback's all model is based on the `PersistedModel`. 

For the question, if want to just simply implement the `created_at` field, it can declared in the `WUser`'s json file, and deal it with the hook `before_save` in `WUser` model.

The reason i make it like this, is based on my pervious experience, for the common and shared field in different model, keep it in the 'parent' can make things easier, but maybe i over-used it in this test.